tests/integrated/test_suite

```diff
@@ -12,53 +12,100 @@ from __future__ import print_function
+
+args = parser.parse_args()
+
+##################################################################
+
 sys.path.append('..')
 from requirements import Requirements
-requirements = Requirements();
-requirements.add('make',False)
+requirements = Requirements()
 
+requirements.add('all_tests', args.all_tests)
 
-##################################################################
+if args.set_bool is not None:
+    lookup = {'false': False,
+              'no': False,
+              'true': True,
+              'yes': True}
+    for setbool in args.set_bool:
+        k, v = setbool.split("=")
+        v = lookup[v.lower()]
+        requirements.add(k, v, override=True)
 
-# Get list of directories containing test cases
-tests=[]
-with open('test_suite_list') as testfile:
-    for test in testfile:
-        test=test.strip()
-        if test.startswith("#"):
-            pass
-        else:
-            tests.append(test)
+try:
+    requirements.add('make', args.make)
+except RuntimeError:
+    raise RuntimeError("The make flag needs to be set by passing --make")
 
 ##################################################################
 
-if "--get-list" in sys.argv:
+if args.set_list is None:
+    # Get list of directories containing test cases
+    try:
+        # Requires python >= 3.5
+        tests = glob.iglob('**/runtest', recursive=True)
+    except TypeError:
+        # Fall back - check only a few folders ...
+        tests = glob.glob('*/runtest')
+        tests += glob.glob('*/*/runtest')
+        tests += glob.glob('*/*/*/runtest')
+
+    tests = [x.rsplit('/', 1)[0] for x in tests]
+
+else:
+    # Take the user provided list
+    tests = [x.rstrip('/') for x in args.set_list]
+
+
+if args.get_list:
     for test in tests:
-        req_met, req_expr = requirements.check(test+"/runtest")
+        req_met, _ = requirements.check(test+"/runtest")
         if req_met:
             print(test)
-    exit(0)
+    sys.exit(0)
+
+##################################################################
+# Run the actual test
+
+command = './runtest' if not args.make else 'make'
 
-savepath = os.getcwd() # Save current working directory
-npassed = 0
-nfailed = 0
-total = len(tests)
+savepath = os.getcwd()  # Save current working directory
 failed = []
 
 start_time = time.time()
 
 print("======= Starting {0} tests ========".format(savepath.split("/")[-1]))
-print("Running {0} tests".format(total))
+if args.make:
+    print("Making {0} tests".format(len(tests)))
+else:
+    print("Running {0} tests".format(len(tests)))
 
-longest=max([len(s) for s in tests])
-output='%-'+str(longest+1)+'s'
+longest = max([len(s) for s in tests])
+output = '%-'+str(longest+1)+'s'
 for t in tests:
     os.chdir(t)
 
-    print(output%t,end='')
-    sys.stdout.flush() # Make sure name is flushed
+    print(output % t, end='')
+    sys.stdout.flush()  # Make sure name is flushed
+
     # Check requirements
     req_met, req_expr = requirements.check("./runtest")
```

Comment:
Should this be `command` rather than `./runtest` as sometimes we just want to make?